stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2889,22 +2945,35 @@ struct _Native${Self}Storage<${TypeParametersDecl}> :
 
   @_versioned
   internal var startIndex: Index {
-    return Index(nativeStorage: self, offset: -1).successor()
+    // This looks weird because of the FIXME in index(after:)
+    return index(after: Index(offset: -1))
   }
 
   @_versioned
   internal var endIndex: Index {
-    return Index(nativeStorage: self, offset: capacity)
+    return Index(offset: capacity)
   }
 
   @_versioned
   internal func index(after i: Index) -> Index {
-    return i.successor()
+    _precondition(i != endIndex)
+    var idx = i.offset + 1
+    while idx < capacity /*&& !nativeStorage[i]*/ {
+      // FIXME: workaround for <rdar://problem/15484639>
```

Comment:
Could you remove the FIXME and the commented-out code?  I don't think the commented-out code still compiles.

You could try `while idx < capacity && isInitializedEntry(at: idx)`, if it works.